### PR TITLE
EditTemplate optimizations

### PR DESCRIPTION
**Pull request by BurningTreeC**
#4093 and #4100 are bundled in this PR

* qualified state-tiddlers for the tags input and fieldname + fieldvalue inputs
* newTagName, newFieldNameTiddler and newFieldValueTiddler variables defined in EditTemplate (all qualified through `qualify` macro)
* save-tiddler-actions macro in the EditTemplate (reused by the save-tiddler button)
* enter (configurable) in the fieldvalue field adds the field and sets focus to the next fieldname input

Edit:

* storyview="pop" for fields list

### CODE REVIEW

**Comment by Jermolene** [comment]
Hi @BurningTreeC this looks good, I'm tagging it for after v5.1.20 to allow for adequate testing.

**Comment by BurningTreeC** [comment]
This is pretty solid and should be merged before #4062

File: core/language/en-GB/EditTemplate.multids [hunk 1]
```diff
@@ -8,6 +8,7 @@ Field/Remove/Hint: Remove field
 Field/Dropdown/Caption: field list
 Field/Dropdown/Hint: Show field list
 Fields/Add/Button: add
+Fields/Add/Button/Hint: Add the fieldname/-value pair to the tiddler
```
Comment (Jermolene) [inline core/language/en-GB/EditTemplate.multids]: This might be clearer as "Add the new field to the tiddler"

File: core/ui/EditTemplate.tid [hunk 2]
```diff
@@ -1,21 +1,22 @@
 title: $:/core/ui/EditTemplate
 
-\define actions()
-<$action-sendmessage $message="tm-add-tag" $param={{$:/temp/NewTagName}}/>
-<$action-deletetiddler $tiddler="$:/temp/NewTagName"/>
-<$action-sendmessage $message="tm-add-field" $name={{$:/temp/newfieldname}} $value={{$:/temp/newfieldvalue}}/>
-<$action-deletetiddler $tiddler="$:/temp/newfieldname"/>
-<$action-deletetiddler $tiddler="$:/temp/newfieldvalue"/>
+\define save-tiddler-actions()
+<$action-sendmessage $message="tm-add-tag" $param={{$(newTagNameTiddler)$}}/>
```
Comment (Jermolene) [inline core/ui/EditTemplate.tid]: Is there any hope of getting rid of the `$()$` constructions here?

Comment (BurningTreeC) [reply core/ui/EditTemplate.tid]: @Jermolene - perhaps by `$param={{ [<newTagNameTiddler>get[text] }}}` ?

Comment (Jermolene) [reply core/ui/EditTemplate.tid]: I think so

Comment (BurningTreeC) [reply core/ui/EditTemplate.tid]: I think it's resolved now

**Comment by BurningTreeC** [comment]
This PR also adds the `pop storyview` to the fields list

**Comment by BurningTreeC** [comment]
@Jermolene, I've refactored the code, it's much cleaner now

**Comment by Jermolene** [comment]
Great thanks @BurningTreeC looks much better.

**Comment by BurningTreeC** [comment]
@Jermolene - this could be included in 5.1.20

**Comment by BurningTreeC** [comment]
this PR could be merged right away for testing

File: core/ui/EditTemplate.tid [hunk 3]
```diff
@@ -1,28 +1,28 @@
 title: $:/core/ui/EditTemplate
 
-\define actions()
-<$action-sendmessage $message="tm-add-tag" $param={{$:/temp/NewTagName}}/>
-<$action-deletetiddler $tiddler="$:/temp/NewTagName"/>
-<$action-sendmessage $message="tm-add-field" $name={{$:/temp/newfieldname}} $value={{$:/temp/newfieldvalue}}/>
-<$action-deletetiddler $tiddler="$:/temp/newfieldname"/>
-<$action-deletetiddler $tiddler="$:/temp/newfieldvalue"/>
+\define save-tiddler-actions()
+<$action-sendmessage $message="tm-add-tag" $param={{{ [<newTagNameTiddler>get[text]] }}}/>
```
Comment (Jermolene) [inline core/ui/EditTemplate.tid]: Apologies if we've covered this before, but why has `{{$:/temp/NewTagName}}` been changed to `{{{ [<newTagNameTiddler>get[text]] }}}`?

Comment (BurningTreeC) [reply core/ui/EditTemplate.tid]: That's because the input tiddlers are qualified tiddlers

**Comment by Jermolene** [comment]
Hi @BurningTreeC is this ready to merge?

**Comment by BurningTreeC** [comment]
@Jermolene, yes this is ready

**Comment by BurningTreeC** [comment]
ping @Jermolene

**Pull request merged**